### ISSUE TITLE
User-driven/configurable algorithmic discovery

### ISSUE THREAD

**Issue by fu5ha** [syuilo/Misskey#9160]
## Summary

For many reasons (I can elaborate if this is not an immediate "no"), I think that some form of algorithmic discovery is desirable, as it is simply impossible for a user to be online enough to sort through all possible content manually. However, algorithms built by large corporations which seek to maximize user engagement have very obvious downsides in many well-discussed ways. Still, I believe there is room for a good kind of algorithm which is *opt-in* and *user-controlled*. Since misskey has no monetary incentive to maximize user engagement, we're instead free to design an algorithm which centers the *user,* allowing said user to more effectively find the content *they* want to find.

It would be really great if this could be configured live in real-time in some fashion, i.e. making it into an intelligent search of some sort.

However, I know that in some places the mere idea of any algorithm being introduced is basically heresy. I searched here and the forum to see if I could find any discussion, but in english and my online-translator powered japanese searches I couldn't find anything about whether the creators of misskey have any stance on this.

**Comment by ThatOneCalculator** [comment]
This already is already in Misskey somewhat with the explore tab, where you can view popular posts and both local & remote users.

**Comment by acid-chicken** [comment]
The ability to delegate recommendations to external services was implemented in [v10.1.0](https://github.com/misskey-dev/misskey/compare/10.0.0...10.1.0) but has been removed as it is no longer maintained from anyone.

**Comment by ThatOneCalculator** [comment]
Oh, I didn't even know that was a thing. I might reimplement that in Calckey 👀

**Comment by acid-chicken** [comment]
It was 4 years ago, so it is no wonder you do not know about it. And it seems that some of the logs are missing because the implementer deleted their account.

**Comment by fu5ha** [comment]
> This already is already in Misskey somewhat with the explore tab, where you can view popular posts and both local & remote users.

Indeed! This is one of the reasons I thought this would be more well received by misskey than some other existing ActivityPub-based services.

Okay, sounds like there's at least room for exploration here.

No promises, but I'll try to draft up some sort of more detailed proposal and hopefully try to move forward with implementing something.

**Comment by joaovl5** [comment]
Hey! This could be a cool feature for the user to be able to choose whether or not he wants algorithmic discovery on their timelines, and allow for multiple algorithm choices. Are there any plans to add this?
